services/ui-src/src/components/fields/DynamicField.test.tsx

```diff
@@ -0,0 +1,50 @@
+import { render, screen } from "@testing-library/react";
+import { axe } from "jest-axe";
+//components
+import { DynamicField } from "components";
+
+const appendButtonText = "Add a row";
+
+jest.mock("react-hook-form", () => ({
+  useFormContext: () => ({
+    register: () => {},
+  }),
+  useFieldArray: jest.fn().mockReturnValue({
+    fields: [
+      {
+        id: "uniqueId",
+        0: {
+          0: "text",
+        },
+      },
+    ],
+    append: () => {},
+    remove: () => {},
+  }),
+}));
+
+const dynamicFieldComponent = (
+  <DynamicField name="testDynamicField" label="test-label" />
+);
+
+describe("Test DynamicField component", () => {
+  test("DynamicField is visible", () => {
+    render(dynamicFieldComponent);
+    const inputBoxLabel = screen.getByText("test-label");
+    expect(inputBoxLabel).toBeVisible();
+  });
+
+  test("DynamicField append button is visible", () => {
+    render(dynamicFieldComponent);
+    const appendButton = screen.getByText(appendButtonText);
+    expect(appendButton).toBeVisible();
+  });
```

Comment:
Need to add a couple more tests here for: 
1. Append button functionality
2. Remove button functionality
3. Remove button only shows on elements after the first.